Dynatrace/dynatrace-operator
# Trying things to fix conversion

## Description
This is all the changes I did in order to find the source of the super annoying error:
```
admission webhook "webhook.dynatrace.com" denied the request: unable to decode dynatrace.com/v1beta1, Kind=DynaKube into *dynakube.DynaKube
```
But in my search of the cause I also did some changes that I think are an improvement OR absolutely necessary.

⚠️This is not a complete, merge and you are done change, the linting and tests have been messed up.

Detailed changes:
- The fix for the admission error:
   - It had nothing really to do with the conversion webhook, the problem was that the `ValidatingWebhookConfiguration` still was asking for `v1beta1` even though the code wanted to work with `v1beta2`, and (I guess) the validation webhook wanted validate the `v1beta1` dynakube  BEFORE the conversion even happened, but failed, as it couldn't convert it on its own.
- Moved the `Hub` to the `v1beta2` version
    - If our code is working only with `v1beta2` it makes no sense to store it in `v1beta1` as that would lead to constant conversions.
       - I know we did it like that for `v1alpha1` but its incorrect, so lets not repeat our mistake.
- Removed all the `operator-sdk` labels from the CR and introduced `kubebuilder` ones
   - The `operator-sdk` labels were just noise, and provided nothing
   - Now we are conforming to how we did it in `EdgeConnect`, yay
   - No more pointer/nil values for simple types, because we can set default values via `kubebuilder`, yaaaay
      - ⚠️ This lead to some code changes
- Removed code from the `v1beta1` package, as they were not used, and shouldn't be used
  - Feautre-flags are an exceptions, as those are needed for conversion (⚠️ the ones that are not converted still need to be removed as they will present in `v1beta2`)
  - Removed the `properties.go`, maybe that was a mistake, could be useful for conversion, but IMO its cleaner if we do the conversion "raw", and nothing is hidden behind magic functions.
- Updated the `ConvertTo` function (⚠️ very important and its not finished, and `ConvertFrom` needs similar type of love)
   -  Before my change, the `ConvertTo/From` function only cared about the conversion that I would consider "more complicated", but it didn't much care for the very annoying an monotonous ones. (like copying all the `v1beta1.CloudNativeSpec` to the 99% same `v1beta2.CloudNativeSpec`) 
   - These super monotonous conversions need to be done, because when you are using the `webhook` conversion method, there is no help from kubernetes, its all in our responsibility to do.
      - There are probably nicer ways to do it, because of how similar the structures are, but I went the safe/least-magic route.



## How can this be tested?


Apply 1 `v1beta1` Dynakube, should work and move most fields (didn't fix the moving of the whole status)
Apply 1 `v1beta2` Dynakube, should just work, didn't touch the conversion logic of that, but didn't have a problem with it as of now
